sphinx_gallery/gen_rst.py

```diff
@@ -188,7 +188,11 @@ def split_code_and_text_blocks(source_file):
         List where each element is a tuple with the label ('text' or 'code'),
         and content string of block.
     """
+    config = dict()
+
     docstring, rest_of_content = get_docstring_and_rest(source_file)
+    rest_of_content, thumbnail_number = extract_thumbnail_number(rest_of_content)
```

Comment:
You don't change rest_of_content anymore so you only need to return the thumbnail number from `extract_thumbnail_number`.